test/integration/scheduler/preemption_test.go

```diff
@@ -355,4 +357,325 @@ var _ = ginkgo.Describe("Preemption", func() {
+				testing.MakeAdmission(aBestEffortCQ.Name).Assignment(corev1.ResourceCPU, "fallback", "5").Obj(),
+			)
+		})
+
+		ginkgo.It("Should not allow a higher-priority workload to preempt a lower-priority workload, if above the threshold", func() {
+			ginkgo.By("Create a high priority workload which requires borrowing")
+			aStandardHighWl := testing.MakeWorkload("a-standard-high", ns.Name).
+				Queue(aStandardLQ.Name).
+				Priority(highPriority).
+				Request(corev1.ResourceCPU, "5").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, aStandardHighWl)).To(gomega.Succeed())
+
+			ginkgo.By("Await for the a-standard-high workload to be admitted")
+			util.ExpectWorkloadToBeAdmittedAs(ctx, k8sClient, aStandardHighWl,
+				testing.MakeAdmission(aStandardCQ.Name).Assignment(corev1.ResourceCPU, "one", "5").Obj(),
+			)
+
+			ginkgo.By("Create the b-standard-very-high workload")
+			bStandardVeryHighWl := testing.MakeWorkload("b-standard-very-high", ns.Name).
+				Queue(bStandardLQ.Name).
+				Priority(veryHighPriority).
+				Request(corev1.ResourceCPU, "5").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, bStandardVeryHighWl)).To(gomega.Succeed())
+
+			ginkgo.By("Verify the b-standard-very-high is admitted using the second flavor, as it couldn't preempt in the first")
+			util.ExpectWorkloadToBeAdmittedAs(ctx, k8sClient, bStandardVeryHighWl,
+				testing.MakeAdmission(bStandardCQ.Name).Assignment(corev1.ResourceCPU, "fallback", "5").Obj(),
+			)
+		})
+
+		ginkgo.It("Should not allow a higher-priority workload to preempt a lower-priority workload, if the low priority one is not borrowing", func() {
+			ginkgo.By("Create a low priority workload which does not borrowing")
+			aBestEffortLowWl := testing.MakeWorkload("a-best-effort-low", ns.Name).
+				Queue(aBestEffortLQ.Name).
+				Priority(lowPriority).
+				Request(corev1.ResourceCPU, "1").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, aBestEffortLowWl)).To(gomega.Succeed())
+
+			ginkgo.By("Await for the a-best-effort-low workload to be admitted")
+			util.ExpectWorkloadToBeAdmittedAs(ctx, k8sClient, aBestEffortLowWl,
+				testing.MakeAdmission(aBestEffortCQ.Name).Assignment(corev1.ResourceCPU, "one", "1").Obj(),
+			)
+
+			ginkgo.By("Create the a-standard-high workload")
+			aStandardHighWl := testing.MakeWorkload("a-standard-high", ns.Name).
+				Queue(aStandardLQ.Name).
+				Priority(veryHighPriority).
+				Request(corev1.ResourceCPU, "8").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, aStandardHighWl)).To(gomega.Succeed())
+
+			ginkgo.By("Verify the a-best-effort-low is admitted using the second flavor, as it couldn't preempt in the first")
+			util.ExpectWorkloadToBeAdmittedAs(ctx, k8sClient, aStandardHighWl,
+				testing.MakeAdmission(aStandardCQ.Name).Assignment(corev1.ResourceCPU, "fallback", "8").Obj(),
+			)
+		})
+	})
+
+	ginkgo.It("Should prioritize scheduling of non-borrowing workloads in the same scheduling cycle", func() {
```

Comment:
I have observed with debugger this test isn't testing what I wanted. The two workloads aren't scheduled in the same scheduling cycle. I will investigate what is going on, but it seems to make sense to remove it for now. We can cover this with unit tests.

Reply:
Same cycle integration test is removed